decoder/expression_candidates_test.go

```diff
@@ -2745,3 +2749,178 @@ variable "ccc" {}
+					return candidates, nil
+				},
+			},
+			lang.IncompleteCandidates([]lang.Candidate{
+				{
+					Label:       "\"insertText\"",
+					Detail:      "detail",
+					Kind:        lang.StringCandidateKind,
+					Description: lang.PlainText("description"),
+					TextEdit: lang.TextEdit{
+						NewText: "\"insertText\"",
+						Snippet: "\"insertText\"",
+						Range: hcl.Range{
+							Filename: "test.tf",
+							Start:    hcl.Pos{Line: 1, Column: 8, Byte: 7},
+							End:      hcl.Pos{Line: 1, Column: 8, Byte: 7},
+						},
+					},
+				},
+			}),
+		},
+		{
+			"hook with prefix",
+			map[string]*schema.AttributeSchema{
+				"attr": {
+					Expr: schema.LiteralTypeOnly(cty.String),
+					CompletionHooks: lang.CompletionHooks{
+						{
+							Name: "TestCompletionHook",
+						},
+					},
+				},
+			},
+			`attr = "pa"
+`,
+			hcl.Pos{Line: 1, Column: 11, Byte: 10},
+			CompletionFuncMap{
+				"TestCompletionHook": func(ctx context.Context, value cty.Value) ([]Candidate, error) {
+					candidates := []Candidate{
+						{
+							Label:         value.AsString(),
+							Kind:          lang.StringCandidateKind,
+							RawInsertText: value.AsString(),
+						},
+					}
+					return candidates, nil
+				},
+			},
+			lang.IncompleteCandidates([]lang.Candidate{
+				{
+					Label: "\"pa\"",
+					Kind:  lang.StringCandidateKind,
+					TextEdit: lang.TextEdit{
+						NewText: "\"pa\"",
+						Snippet: "\"pa\"",
+						Range: hcl.Range{
+							Filename: "test.tf",
+							Start:    hcl.Pos{Line: 1, Column: 8, Byte: 7},
+							End:      hcl.Pos{Line: 1, Column: 12, Byte: 11},
+						},
+					},
+				},
+			}),
+		},
+		{
+			"incomplete attr value",
+			map[string]*schema.AttributeSchema{
+				"attr": {
+					Expr: schema.LiteralTypeOnly(cty.String),
+					CompletionHooks: lang.CompletionHooks{
+						{
+							Name: "TestCompletionHook",
+						},
+					},
+				},
+			},
+			`attr = "pa
+
+`,
+			hcl.Pos{Line: 1, Column: 11, Byte: 10},
+			CompletionFuncMap{
+				"TestCompletionHook": func(ctx context.Context, value cty.Value) ([]Candidate, error) {
+					candidates := []Candidate{
+						{
+							Label:         value.AsString(),
+							Kind:          lang.StringCandidateKind,
+							RawInsertText: value.AsString(),
+						},
+					}
+					return candidates, nil
+				},
+			},
+			lang.IncompleteCandidates([]lang.Candidate{
+				{
+					Label: "\"pa\"",
+					Kind:  lang.StringCandidateKind,
+					TextEdit: lang.TextEdit{
+						NewText: "\"pa\"",
+						Snippet: "\"pa\"",
+						Range: hcl.Range{
+							Filename: "test.tf",
+							Start:    hcl.Pos{Line: 1, Column: 8, Byte: 7},
+							End:      hcl.Pos{Line: 1, Column: 11, Byte: 10},
+						},
+					},
+				},
+			}),
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("%d-%s", i, tc.testName), func(t *testing.T) {
+			bodySchema := &schema.BodySchema{
+				Attributes: tc.attrSchema,
+			}
+
+			f, _ := hclsyntax.ParseConfig([]byte(tc.cfg), "test.tf", hcl.InitialPos)
```

Comment:
Nitpick: A comment to explain why we're ignoring the diagnostics here would be useful (i.e. that we expect invalid configs), just to decrease the chance of someone just blindly copy pasting it elsewhere where this context is not applicable.

Reply:
I just blindly copy pasted it from above 😇